File: Advent Of Code/2023/Day 1/day-1.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
#include <cctype>
#include <cstdlib>
using namespace std;


int extract_number(string s) {
    char first = '\0', second = '\0';
    for (auto i = s.begin(); i != s.end(); i++) {
        if (isdigit(*i)) {
            if (first == '\0') {
                first = *i;
                second = *i;
            } else {
                second = *i;
            }
        }
    }
    return (first - '0') * 10 + (second - '0');
}



int main() {
    // ifstream input ("sample-input.txt", ifstream::in);
    ifstream input ("puzzle-input.txt", ifstream::in);
    string current;
    int sum = 0;

    while (input >> current) {
        sum += extract_number(current);
    }
    cout << sum << endl;
}

File: Advent Of Code/2023/Day 1/day-1.py
def extract_number(s):
    first = None
    second = None

    for c in s:
        if c.isdigit():
            if first is None:
                first = c
                second = c
            else:
                second = c
    
    return int(first) * 10 + int(second)


def main():
    total_sum = 0
    # with open("sample-input.txt", "r") as fin:
    with open("puzzle-input.txt", "r") as fin:
        for line in fin:
            line = line.strip()
            if line:
                total_sum += extract_number(line)
    
    print(total_sum)


if __name__ == "__main__":
    main()
